src/altinn-app-frontend/src/common/hooks/useFormLayoutHistory.ts

```diff
@@ -0,0 +1,61 @@
+import { useCallback, useEffect } from 'react';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+
+import { useAppDispatch } from 'src/common/hooks/useAppDispatch';
+import { FormLayoutActions } from 'src/features/form/layout/formLayoutSlice';
+import type { IUpdateCurrentView } from 'src/features/form/layout/formLayoutTypes';
+
+/**
+ * @param activePageId the formLayout page id that is currently active in the app
+ * @return matchRootUrl: Matches the location pathname of the history root. If not, an empty string.
+ */
+export const useFormLayoutHistoryAndMatchInstanceLocation = ({
+  activePageId,
+}: {
+  activePageId: string;
+}): { matchRootUrl: string } => {
+  const matchRoot = useRouteMatch<string>('/instance/:partyId/:instanceGuid');
+  const matchRootUrl = matchRoot?.url || '';
+  const history = useHistory();
+  const dispatch = useAppDispatch();
+  const dispatchAction = FormLayoutActions.updateCurrentView;
+  const doDispatch = useCallback(
+    (payload: IUpdateCurrentView) => {
+      dispatch(dispatchAction(payload));
+    },
+    [dispatch, dispatchAction],
+  );
```

Comment:
I don't see the point in declaring a new dispatch function here. You don't have to pass `dispatchAction` as a dependency to `useCallback`, it's fine just calling `dispatch(FormLayoutActions.updateCurrentView({...}))` directly. With all of this indirection it just makes the code harder to follow, IMO.